docs/developers/blobstream-proof-query.md

```diff
@@ -0,0 +1,367 @@
+
+### Data root inclusion proof
+
+To prove the data root is committed to by the Blobstream smart contract, we will need to provide a Merkle proof of the data root tuple to a data root tuple root. This can be created using the [`data_root_inclusion_proof`](https://github.com/celestiaorg/celestia-core/blob/c3ab251659f6fe0f36d10e0dbd14c29a78a85352/rpc/client/http/http.go#L492-L511) query.
+
+This [endpoint](https://github.com/celestiaorg/celestia-core/blob/793ece9bbd732aec3e09018e37dc31f4bfe122d9/rpc/openapi/openapi.yaml#L1045-L1093) allows querying data root to data root tuple root proof. It takes a block `height`, a starting block and an end block, then it generates the binary merkle proof of the `DataRootTuple`, corresponding to that `height`, to the `DataRootTupleRoot` which is committed to in the Blobstream contract.
+
+Example request: `/data_root_inclusion_proof?height=15&start=10&end=20`
+
+Which queries the proof of the height `15` to the data commitment defined by the range `[10, 20)`.
+
+Example response:
+
+```json
+{
+   "jsonrpc":"2.0",
+   "id":-1,
+   "result":{
+      "proof":{
+         "total":"10",
+         "index":"5",
+         "leaf_hash":"vkRaRg7FGtZ/ZhsJRh/Uhhb3U6dPaYJ1pJNEfrwq5HE=",
+         "aunts":[
+            "nmBWWwHpipHwagaI7MAqM/yhCDb4cz7z4lRxmVRq5f8=",
+            "nyzLbFJjnSKOfRZur8xvJiJLA+wBPtwm0KbYglILxLg=",
+            "GI/tJ9WSwcyHM0r0i8t+p3hPFtDieuYR9wSPVkL1r2s=",
+            "+SGf6MfzMmtDKz5MLlH+y7mPV9Moo2x5rLjLe3gbFQo="
+         ]
+      }
+   }
+}
+```
+
+> **_NOTE:_** The values are base64 encoded. For these to be usable with the solidity smart contract, they need to be converted to `bytes32`. Check the next section for more information.
+
+### 2. Transaction inclusion proof
+
+To prove that a rollup transaction is part of the data root, we will need to provide two proofs: a namespace Merkle proof of the transaction to a row root. This could be done via proving the shares that contain the transaction to the row root using a namespace Merkle proof. And, a binary Merkle proof of the row root to the data root.
+
+These proofs can be generated using the [`ProveShares`](https://github.com/celestiaorg/celestia-core/blob/c3ab251659f6fe0f36d10e0dbd14c29a78a85352/rpc/client/http/http.go#L526-L543) query.
+
+This [endpoint](https://github.com/celestiaorg/celestia-core/blob/793ece9bbd732aec3e09018e37dc31f4bfe122d9/rpc/core/tx.go#L175-L213) allows querying a shares proof to row roots, then a row roots to data root proofs. It takes a block `height`, a starting share index and an end share index which define a share range. Then, two proofs are generated:
+
+- An NMT proof of the shares to the row roots
+- A binary merkle proof of the row root to the data root
+
+> **_NOTE:_** if the share range spans multiple rows, then the proof can contain multiple NMT and binary proofs.
+
+Example request: `/prove_shares?height=15&startShare=0&endShare=1`
+
+Which queries the proof of shares `[0,1)` in block `15`.
```

Comment:
should this be a ]?

Reply:
this is meant to be end exclusive. I used `[` in documentation before but apparently `)` is more used to signify end exclusivity